src/app/digital/views/DigitalPortView.ts

```diff
@@ -25,4 +37,26 @@ export class DigitalPortView extends PortView<DigitalPort, DigitalViewInfo> {
             (this.obj.group !== "outputs" && (p.group === "outputs"))     // TODOnow: fix
         );
     }
+
+    protected override renderInternal({ renderer, selections }: RenderInfo): void {
+        const parentSelected = selections.has(this.obj.parent);
+        const selected = selections.has(this.obj.id);
+
+        const { origin, target } = this.pos.get();
+
+        const lineCol       = (parentSelected && !selected ? SELECTED_BORDER_COLOR : DEFAULT_BORDER_COLOR);
+        const borderCol     = (parentSelected ||  selected ? SELECTED_BORDER_COLOR : DEFAULT_BORDER_COLOR);
+        const circleFillCol = (parentSelected ||  selected ? SELECTED_FILL_COLOR   : DEFAULT_FILL_COLOR);
+        const lineStyle   = new Style(undefined, lineCol, IO_PORT_LINE_WIDTH);
+        const circleStyle = new Style(circleFillCol, borderCol, IO_PORT_BORDER_WIDTH);
+
+        renderer.draw(new Line(origin, target), lineStyle);
+        renderer.draw(new Circle(target, IO_PORT_RADIUS), circleStyle);
+
+        if (this.circuit.getPortParent(this.obj).kind === "NANDGate" && this.obj.group === "outputs"){
```

Comment:
make a constant right above called `parent = this.circuit.getPortParent(this.obj)` to simplify this a bit
Also add a comment explaining why this is necessary for now